File: CPP Assignment 2/Bank_System_Modular/BankAccount.cpp
//============================================================================
// Name        : BankSystem.cpp
// Author      : 
// Version     :
// Copyright   : Your copyright notice
// Description : Hello World in C++, Ansi-style
//============================================================================

#include "bankaccount.h"

BankAccount::BankAccount() {
    accountNumber = 0;
    accountHolderName = "";
    balance = 0;
}

void BankAccount::setAccountNumber(int acc) {
    accountNumber = acc;
}

void BankAccount::setAccountHolderName(string name) {
    accountHolderName = name;
}

int BankAccount::getAccountNumber() {
    return accountNumber;
}

string BankAccount::getAccountHolderName() {
    return accountHolderName;
}

double BankAccount::getBalance() {
    return balance;
}

void BankAccount::deposit(double amount) {
    if (amount > 0) {
        balance += amount;
        cout << "Deposit Successful!\n";
    } else {
        cout << "Invalid Deposit Amount!\n";
    }
}

void BankAccount::withdraw(double amount) {
    if (amount > balance) {
        cout << "Insufficient Balance!\n";
    } else {
        balance -= amount;
        cout << "Withdrawal Successful!\n";
    }
}

void BankAccount::displayAccountDetails() {
    cout << "\nAccount No: " << accountNumber << endl;
    cout << "Name: " << accountHolderName << endl;
    cout << "Balance: " << balance << endl;
}

File: OOPJ Assignments/Assignment 2/BankAccount.java
class BankAccount {
    int accountNumber;
    String name;
    double balance;

    BankAccount() {
        accountNumber = 0;
        name = "Unknown";
        balance = 0.0;
    }

    BankAccount(int accountNumber, String name, double balance) {
        this.accountNumber = accountNumber;
        this.name = name;
        this.balance = balance;
    }

    void display() {
        System.out.println(accountNumber + " " + name + " " + balance);
    }

    public static void main(String[] args) {
        BankAccount b1 = new BankAccount();
        BankAccount b2 = new BankAccount(101, "Raj", 5000);
        BankAccount b3 = new BankAccount(102, "Mohit", 8000);

        System.out.println("Account Details:");
        b1.display();
        b2.display();
        b3.display();
    }
}